src/Illuminate/Collections/Collection.php

```diff
@@ -1050,6 +1052,31 @@ public function splitIn($numberOfGroups)
         return $this->chunk(ceil($this->count() / $numberOfGroups));
     }
 
+    /**
+     * Get the first item in the collection, but only if exactly
+     * item exists. Otherwise, throw an exception.
+     *
+     * @param  callable|null  $callback
+     * @return mixed
+     *
+     * @throws ItemNotFoundException
+     * @throws MultipleItemsFoundException
+     */
+    public function sole(callable $callback = null)
+    {
+        $items = $this->filter($callback);
```

Comment:
The code as is will filter out `null`s when no `$callback` has been provided.

```suggestion
        $items = $this->when($callback)->filter($callback);
```

Reply:
Ahh I must have missed that, cheers